google/cloud/spanner/internal/clock.h

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_SPANNER_GOOGLE_CLOUD_SPANNER_INTERNAL_CLOCK_H
+#define GOOGLE_CLOUD_CPP_SPANNER_GOOGLE_CLOUD_SPANNER_INTERNAL_CLOCK_H
+
+#include "google/cloud/spanner/version.h"
+#include <chrono>
+#include <functional>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+namespace internal {
+
+/// Until we need something more complicated, use `steady_clock` as our Clock.
+using Clock = std::chrono::steady_clock;
+
+/// A function type that is expected to yield a `time_point` representing "now".
+using ClockNowSource = std::function<Clock::time_point()>;
```

Comment:
We already live in a world, with our retry and polling loops(*), where non-test code wants to "sleep" until the clock reaches some point (or an interval has passed).  So, I would suggest that we should support a richer interface than just "now".

[*] At the moment we have to override the policy parameters to avoid long-blocking tests.